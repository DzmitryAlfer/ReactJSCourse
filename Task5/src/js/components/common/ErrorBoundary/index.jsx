import React, { PureComponent } from 'react';
import * as actions from "../../../actions";
import {connect} from "react-redux";

class ErrorBoundary extends PureComponent {
    componentDidCatch(error, info) {
        this.props.dispatch(actions.app.setError(true));
    }

    render(){
        if(this.props.hasError) {
            return (<h1>Ooops, Something went wrong</h1>);
        }

        return this.props.children;
    }
}

const mapStateToProps = state => {
    const { appReducer } = state;

    return {
        hasError : appReducer.hasError
    }
};

export default connect (mapStateToProps)(ErrorBoundary);