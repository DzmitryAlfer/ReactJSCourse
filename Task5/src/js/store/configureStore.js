import { createStore, applyMiddleware, compose } from 'redux'
import { createLogger } from 'redux-logger'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer)

export default function configureStore(initialState) {
    const sagaMiddleware = createSagaMiddleware()

    const store = createStore(
        persistedReducer,
        initialState,
        compose(
            applyMiddleware(
                sagaMiddleware,
                createLogger()
            )
        )
    );
    const persistor = persistStore(store);

    /*if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextRootReducer = require('../reducers').default;
            store.replaceReducer(nextRootReducer);
        });
    }*/

    store.runSaga = sagaMiddleware.run;
    store.close = () => store.dispatch(END);
    return {store, persistor};
}