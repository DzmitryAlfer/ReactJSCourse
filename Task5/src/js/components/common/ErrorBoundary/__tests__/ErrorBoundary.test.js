import React from 'react';
import Renderer from 'react-test-renderer';
import ErrorBoundary from "../";
import { Provider } from 'react-redux'
import configureStore from "../../../../store/configureStore";

const {store} = configureStore();

describe('ErrorBoundary', () => {
    test('Component render without exception', () => {
        const tree = Renderer.create(
            <Provider store={store}>
                <ErrorBoundary>
                    <span>This text should be rendered</span>
                </ErrorBoundary>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    /*test('Component render with exception', () => {

        const thr = () => { throw 'EXCEPTION'};

        const tree = Renderer.create(
            <ErrorBoundary>
                { 1/0 }
            </ErrorBoundary>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });*/
});