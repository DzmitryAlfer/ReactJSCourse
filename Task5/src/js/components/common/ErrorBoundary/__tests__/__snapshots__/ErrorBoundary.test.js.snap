// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component render without exception 1`] = `
<span>
  This text should be rendered
</span>
`;
